### PR TITLE
ringing-lib: init at 0-unstable-2024-05-31

### PR DESCRIPTION
**Pull request by jshholland**
###### Description of changes

Tools and C++ class library for English change ringing: http://ringing-lib.github.io/

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/ringing-lib/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{ stdenv, fetchFromGitHub, lib, autoreconfHook, pkg-config, readline, xercesc }:
+stdenv.mkDerivation rec {
+  pname = "ringing-lib";
+  version = "0.4.0-pre";
```
Comment (FliegendeWurst) [inline pkgs/applications/misc/ringing-lib/default.nix:4]:
```suggestion
  version = "unstable-2022-12-15";
```

File: pkgs/applications/misc/ringing-lib/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+  nativeBuildInputs = [ autoreconfHook pkg-config ];
+  buildInputs = [ readline xercesc ];
+
+  meta = with lib; {
+    description = "Library of C++ classes and utilities for change ringing";
+    homepage = "https://ringing-lib.github.io/";
+    license = with licenses; [ gpl2Plus lgpl2Plus ];
+    maintainers = with maintainers; [ jshholland ];
+  };
```
Comment (FliegendeWurst) [inline pkgs/applications/misc/ringing-lib/default.nix:21]:
```suggestion
    platforms = platforms.linux;
  };
```
Doesn't build successfully on MacOS.

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

**Comment by jshholland** [comment]
Applied suggested fixes, rebased & moved to by-name, updated package version and set `doCheck = true`.

**Review by FliegendeWurst — APPROVED** [review]
Result of `nixpkgs-review pr 210560` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ringing-lib</li>
  </ul>
</details>

**Comment by JohnRTitor** [comment]
Result of `nixpkgs-review pr 210560` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ringing-lib</li>
  </ul>
</details>

File: pkgs/by-name/ri/ringing-lib/package.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "ringing-lib";
+  version = "unstable-2023-01-14";
+
+  src = fetchFromGitHub {
+    owner = finalAttrs.pname;
+    repo = finalAttrs.pname;
+    rev = "2649ca69f24b531cfc67027898c2db2cc778a394";
+    sha256 = "sha256-Lm9lWH2LTKm1NX34BVa0P0k2Qsz9JUqgNPM6roGmNss=";
```
Comment (JohnRTitor) [inline pkgs/by-name/ri/ringing-lib/package.nix:11]:
```suggestion
    owner = "ringing-lib";
    repo = "ringing-lib";
    rev = "2649ca69f24b531cfc67027898c2db2cc778a394";
    hash = "sha256-Lm9lWH2LTKm1NX34BVa0P0k2Qsz9JUqgNPM6roGmNss=";
```

File: pkgs/by-name/ri/ringing-lib/package.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  };
+
+  nativeBuildInputs = [ autoreconfHook pkg-config ];
+  buildInputs = [ readline xercesc ];
+
+  doCheck = true;
+
```
Comment (JohnRTitor) [inline pkgs/by-name/ri/ringing-lib/package.nix:18]:
```suggestion
  passthru.updateScript = unstableGitUpdater { };
  
```

File: pkgs/by-name/ri/ringing-lib/package.nix [hunk 5]
```diff
@@ -0,0 +1,26 @@
+{ stdenv, fetchFromGitHub, lib, autoreconfHook, pkg-config, readline, xercesc }:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "ringing-lib";
+  version = "unstable-2023-01-14";
```
Comment (JohnRTitor) [inline pkgs/by-name/ri/ringing-lib/package.nix:5]: let's update to the latest commit

**Comment by jshholland** [comment]
Thanks @JohnRTitor for digging this one out from the past!

**Pull request merged**